ngraph/python/src/openvino/inference_engine/ie_api.py

```diff
@@ -84,3 +84,10 @@ def __new__(cls, tensor_desc : TensorDesc, arr : np.ndarray = None):
             return TBlobUint8(tensor_desc, arr, arr_size)
         else:
             raise AttributeError(f'Unsupported precision {precision} for Blob')
+
+def blob_from_file(path_to_bin_file : str) -> BlobWrapper:
+    array = np.fromfile(path_to_bin_file,dtype=np.uint8)
```

Comment:
I'm not sure that weight's precision is always equal to "U8". Maybe one more parameter for this function should be added like `precision` with set default value to "U8" and, please, use the precision_map for getting numpy's dtype (the precision_map you can find above)

Reply:
You should look at _openvino/inference-engine/src/inference_engine/ie_network_reader.cpp_ around the **line 217**.
`Blob::Ptr weights = make_shared_blob<uint8_t>({Precision::U8, { fileSize }, C });`
Looks like it is forcing it to U8 by default.

Of course we can use `dtype` attribute in `blob_from_file` and set it to default but I do not see a code responsible for other possible precisions inside IE.

Reply:
Ok, I discussed this with relevant people - a blob over weights can be created with U8 precision.